src/shared/Core/Settings.cs

```diff
@@ -334,6 +336,28 @@ public IEnumerable<string> GetSettingValues(string envarName, string section, st
             {
                 IGitConfiguration config = _git.GetConfiguration();
 
+                // Enumerate all configuration entries with the correct section and property name
+                // and make a local copy of them here to avoid needing to call `TryGetValue` on the
+                // IGitConfiguration object multiple times in a loop below.
+
+                // This is a performance optimisation to avoid calling `TryGet` on the IGitConfiguration
+                // object multiple times in a loop below, or each time this entire method is called.
+                // The assumption is that the configuration entries will not change between invocations
+                // of Git Credential Manager, which is reasonable given process lifetime is typically less
+                // than a few seconds. For some entries (type=path), we still need to ask Git in order to
+                // expand the path correctly.
+                if (_configEntries is null)
+                {
+                    _configEntries = new Dictionary<string, string>(GitConfigurationKeyComparer.Instance);
```

Comment:
Is there any chance of concurrent calls to `GetSettingValues()`? If not, this is fine; otherwise, this would probably need some kind of read/write locking setup.

Reply:
Right now almost nothing of the GCM shared components are thread-safe.